cli/src/merge_tools/mod.rs

```diff
@@ -435,6 +471,22 @@ pub fn generate_diff(
     right_tree: &Tree,
     matcher: &dyn Matcher,
     tool: &MergeTool,
+) -> Result<(), DiffGenerateError> {
+    match tool {
+        MergeTool::Internal => unimplemented!("generate_diff with internal mergetool"),
+        MergeTool::External(tool) => {
+            generate_diff_external(tool, left_tree, right_tree, matcher, writer)?;
+        }
+    }
+    Ok(())
+}
+
+fn generate_diff_external(
```

Comment:
Perhaps, the original function can be just renamed to `generate_diff_external()`.

The "internal" diff implementations exist in `diff_utils.rs`.

Reply:
I kept it as `generate_diff` (since there is no `_internal` variant), but moved it to `external.rs` since that's where all the external helper functions are.